lib/api/2.0/schemas2.js

```diff
@@ -26,26 +28,25 @@ var schemasIdGet = controller(function(req) {
 
 // GET /api/2.0/schemas/tasks
 var taskSchemasGet = controller(function () {
-    return taskOptionValidator.getAllSchemaNames({ includeNameSpace: false });
+    return workflowApiService.getTaskDefinitions()
+    .then(function(tasks) {
```

Comment:
I like `Promise.try` as well, it changes a lot of things (synchronous code or erros, non-bluebird promises) into bluebird promise, though a little cumbersome sometimes.